docs/rfc/active/tags/README.md

```diff
@@ -0,0 +1,174 @@
+business logic (e.g. sensitivity tag on a data set) or more ad hoc use cases, such as a data analyst's favorite
+dashboard. A general tag implementation will also allow us to define and search based on a new kind relationship, like
+which datasets and ML models are marked to contain PII data.
+
+> needs work
+
+## Basic example
+
+> If the proposal involves a new or changed API, include a basic code example. Omit this section if it's not applicable.
+
+## Motivation
+
+> Why are we doing this? What use cases does it support? What is the expected outcome?
+>
+> Please focus on explaining the motivation so that if this RFC is not accepted, the motivation could be used to develop
+> alternative solutions. In other words, enumerate the constraints you are trying to solve without coupling them too
+> closely to the solution you have in mind.
+
+## Requirements
+
+> What specific requirements does your design need to meet? This should ideally be a bulleted list of items you wish to
+> achieve with your design. This can help everyone involved (including yourself!) make sure your design is robust enough
+> to meet these requirements.
+>
+> Once everyone has agreed upon the set of requirements for your design, we can use this list to review the detailed
+> design.
+
+### Extensibility
+
+> Please also call out extensibility requirements. Is this proposal meant to be extended in the future? Are you adding a
+> new API or set of models that others can build on in later? Please list these concerns here as well.
+
+## Non-Requirements
+
+> Call out things you don't want to discuss in detail during this review here, to help focus the conversation. This can
+> include things you may build in the future based off this design, but don't wish to discuss in detail, in which case
+> it may also be wise to explicitly list that extensibility in your design is a requirement.
+>
+> This list can be high level and not detailed. It is to help focus the conversation on what you want to focus on.
+
+## Detailed design
+
+We want to introduce some new under `datahub/metadata-models/src/main/pegasus/com/linkedin/common/`.
+
+### `Tag` entity
+
+First we create a `TagMetadata` entity, which defines the actual tag-object.
+
+The edit property defines the edit rights of the tag, as some tags (like sensitivity tags) should be read-only for a
+majority of users
+
+```
+/**
+ * Tag information
+ */
+record TagMetadata {
+
+  /**
+   * Tag URN, e.g. urn:li:tag:<name>
+   */
+  tag: Urn
+
+   /**
+   * Audit stamp associated with creation of this tag
+   */
+   createStamp: AuditStamp
+}
+```
+
+### `TagEmployment`
+
+We define a `TagEmployment`-model, which describes the application of a tag to a entity
+
+```
+/**
+ * Tag information
+ */
+record TagEmployment {
+
+  /**
+   * Tag in question
+   */
+  tag: TagMetadata
+
+  /**
+   * Who has edit rights to this employment. WIP
+   */
+  edit: union[None, any, role-urn]
```

Comment:
I like leaving this in the proposal, but lets keep an * here until we land on an access-control system for datahub as a whole. I think that will have to be a second parallel RFC, but hopefully we can push access control and tags forward together.

Reply:
Add a comment on this!